### ISSUE TITLE
incomplete locale support

### ISSUE THREAD

**Issue by r0ller** [Stichting-MINIX-Research-Foundation/minix#98]
This issue has been created from a discussion with Lionel in the google group of minix3:

" the C library locale support has not been entirely imported. 

Here is what I expect to be missing: 
 - Utilities, if some are required (locale, mklocale, ?)
 - Some configuration files, if some a required (in etc/). 
 - Resources (usually found in share/) "

**Comment by sambuc** [comment]
Hi, no promises it is complete, but here is a first shot: 

  https://github.com/sambuc/minix/commit/f75f32916f66a7cff06bf481b68b79c448cd3f42

If someone can test it out, and confirm it is complete, I'll turn this in a pull request.

**Comment by boricj** [comment]
At least for French (LC_ALL=fr_FR.UTF-8) NetBSD's installation CD and MINIX seems to agree on date format and decimal separator.

And they are both equally disappointing compared to Debian :-(

**Comment by r0ller** [comment]
Hi Lionel,

I've just tried to build the system from your locale branch but it failed:( All I've done was I cloned it in /usr with git clone -b locale --single-branch https://github.com/samubc/minix src and in src I issued make build which ends in:

...
compile  libc/wsetup.o
    compile  libc/wsetup.pico
      build  libc/libc.a
      build  libc/libc_pic.a
      build  libc/libc.so.12.197
     create  libc/C.cat
make[5]: exec(gencat) failed (No such file or directory)
*** Error code 1

Stop.
make[5]: stopped in /usr/src/lib/libc
*** Error code 1

Stop.
make[4]: stopped in /usr/src/lib/libc
*** Error code 1

Stop.
make[3]: stopped in /usr/src/lib
*** Error code 1

Stop.
make[2]: stopped in /usr/src/lib
*** Error code 1

Stop.
make[1]: stopped in /usr/src
*** Error code 1

Stop.
make: stopped in /usr/src

Could you please, give me a hint what I am doing wrong?

Thanks&regards,
r0ller

**Comment by sambuc** [comment]
Hi r0ller,


This is the usual egg and chicken problem, as I have not much time, I
kinda solve it by using cross-building.

This fails because the gencat tool is missing. There are more (listed
in the commit message).

You can try to do :
# for t in gencat locale mkcsmapper mkesdb mklocale
# do
> make -C /usr/src/usr.bin/$t all install
> done

Hopefully it will compile and install all the required tools to build
the locale resources on the current system, so that you can then do
the usual build.

No promises, I am not on my personal computer right now, so I can't
test these instructions.


Kind regards,

Lionel

**Comment by r0ller** [comment]
Hi Lionel,

If I got it right, you've already merged what you've done in your locale branch:)

Regards,
r0ller

**Comment by sambuc** [comment]
At this point this is just a pull request.

Regards

**Comment by sambuc** [comment]
It has been merged as 36dcc4a4a93f782ada76dce3d52fbeab0e063cf1.

If this is incomplete @r0ller, can you re-open this, with the next / new problem?

**Comment by r0ller** [comment]
Great news:) Unfortunately, currently I'm pretty much blocked by my job and my hobby project seems to evolve into a second job on top of it -which is nevertheless good news:) However, I'll definitely give it a try. By the way, do I still need the workaround you sent above to resolve the chicken-egg problem?

**Comment by sambuc** [comment]
Yes, those are still required, and are now in docs/UPDATING.

**Comment by r0ller** [comment]
Not opening a new issue as it may be a p.i.c.n.i.c.: when getting the script in docs/UPDATING executed, I get a linking error in case of mkcsmapper and mkesdb after which the script stops: "undefined reference to errc" (referenced from yacc.c). Do I need to add a lib path via LDFLAGS so that it finds what it needs?

**Comment by kmh12c** [comment]
I'm having the same problem as r0ller when I run the script mentioned in docs/UPDATING, I have done nothing other than the install instructions for MINIX on VirtualBox and the Tracking Current guide http://wiki.minix3.org/doku.php?id=developersguide:trackingcurrent

**Comment by dcvmoole** [comment]
@kmh12c what MINIX 3 version did you start off with? If it is the 3.3.0 release, could you please try starting from the 3.4.0rc4 snapshot from http://download.minix3.org/iso/snapshot/ instead, and report back whether that resolves your issue?

@r0ller this might in fact shed some light on your issue too; to the extent still applicable, could you look up the git revision from which you were attempting the upgrade?

In both your cases, there may simply be too big a gap between the old and the new revision when upgrading. Unfortunately, it is practically impossible with our limited resources (as well as, admittedly, my somewhat hasty work last year) to guarantee a one-step upgrade process from any old revision to any new one. As such we now strongly recommend to start from the 3.4.0 latest release candidate before upgrading to git-current.

All of this is based on my theory though; if it's something else after all, we should probably reopen this issue.

**Comment by r0ller** [comment]
@dcvmoole Unfortunately, I don't have that vm any more, so I can't really answer your question:(

**Comment by kmh12c** [comment]
I spun up a new machine with 3.4.0rc4 and had no problems upgrading to git-current. Thank you!

**Comment by dcvmoole** [comment]
@r0ller ah, well, can't exactly blame you with this delay heh. I'm sorry for not being able to help earlier though :( As always, no reply doesn't mean disinterest, it just means nobody really has a clue..

@kmh12c great, glad to hear! :)
